File: 3-Strings/1551-Frase-Completa/1551.c
#include <stdio.h>
#include <stdbool.h>
#include <string.h>

int main() {
    bool alpha[26];
    int cases, counter;
    char text[1002];
    scanf ("%d", &cases);
    getchar();
    for (int i = 0; i < cases; ++i) {
        for (int q = 0; q < 26; ++q) {
            alpha[q] = false;
        }
        fgets (text, 1002, stdin);
        text[strcspn(text, "\n")] = '\0';
        for (int q = 0; q < strlen(text); ++q) {
            if (text[q] != ' ' && text[q] != ',') {
                alpha[text[q]-97] = true;
            }
        }
        counter = 0;
        for (int q = 0; q < 26; ++q) {
            if (alpha[q]) {
                ++counter;
            }
        }
        if (counter == 26) {
            printf ("frase completa\n");
        }
        else if (counter >= 13) {
            printf ("frase quase completa\n");
        }
        else {
            printf ("frase mal elaborada\n");
        }
    }
    return 0;
}

File: 3-Strings/1551-Frase-Completa/1551.java
import java.util.Scanner;
public class Main {
	public static void main (String[] args) {
		Scanner scan = new Scanner(System.in);
		char[] text = new char[1000];
		boolean[] alpha = new boolean[26];
		int cases = scan.nextInt(), sum;
        scan.nextLine();
		for (int i = 0; i < cases; ++i) {
			for (int q = 0; q < 26; ++q) {
				alpha[q] = false;
			}
			text = scan.nextLine().toCharArray();
			for (int q = 0; q < text.length; ++q) {
				if (text[q] != ' ' && text[q] != ',') {
					alpha[(int)text[q]-97] = true;
				}
			}
			sum = 0;
			for (int q = 0; q < 26; ++q) {
				if (alpha[q]) {
					++sum;
				}
			}
			if (sum == 26) {
				System.out.println("frase completa");
			}
			else if (sum >= 13) {
				System.out.println("frase quase completa");
			}
			else {
				System.out.println("frase mal elaborada");
			}
		}
	}
}